### DISCUSSION TITLE
Save Function

### DISCUSSION THREAD

**Discussion by kmbasha** [immense/Remotely#915] [General]
The installation was smooth on my Docker instance, but the application does not respond to any 'save' features throughout. It seems to be a bug! Furthermore, I installed the Windows agent and tried to access the terminal from the server portal, but it didn't work either! Any suggestions?

**Comment by bitbound** [comment]
This is likely due to websockets not being handled properly by the reverse proxy.  You can confirm this by opening the dev tools in the browser and watching the console.  Within 30 seconds or so, you'd see a timeout or similar error.

**Reply by kmbasha** [reply]
Any specific setting for nginx handling websockets ?

**Reply by bitbound** [reply]
There's an example [here](https://github.com/immense/Remotely/blob/master/Utilities/Example_Nginx_Config.txt), but it's outdated.

The websocket endpoints are at `/_blazor`, `/hubs/desktop`, `/hubs/viewer`, and `/hubs/service`.  If you have anything else in front of your reverse proxy, that needs to forward headers and handle websocket upgrades correctly too.

I'd recommend using Caddy, as all the defaults work without any configuration.
